pmd-core/src/main/java/net/sourceforge/pmd/util/StringUtil.java

```diff
@@ -25,6 +25,48 @@ private StringUtil() {
     }
 
     /**
+     * Formats a double to a percentage, keeping {@code numDecimal} decimal places.
+     *
+     * @param val         a double value between 0 and 1
+     * @param numDecimals The number of decimal places to keep
+     * @param truncateInt Whether to truncate the string to an int if possible (if this is set, then 10.0% will be
+     *                    represented as "10%")
+     *
+     * @return A formatted string
+     *
+     * @throws IllegalArgumentException if the double to format is not between 0 and 1
+     */
+    public static String percentageString(double val, int numDecimals, boolean truncateInt) {
+        if (val < 0 || val > 1) {
+            throw new IllegalArgumentException("Expected a number between 0 and 1");
+        }
+
+        return truncateDouble(100 * val, numDecimals, truncateInt) + "%";
```

Comment:
That's nice indeed! We'll have to override the locale though, the decimal separator is locale-specific (for me it's a comma). Tests that pass on my machine could fail on other machines.

Truncate to int is only really needed for the tests, I think it actually looks neater/ more consistent with decimals